In [1]:
import pandas as pd
import re

In [2]:
!pip install jupysql --upgrade

In [3]:
df = pd.read_csv(
    r"C:\Users\paulo\Desktop\pythonAnalise\analise\projeto3\Netflix_2016_2025.csv", 
    encoding='utf-8', 
    encoding_errors='replace'
)



In [4]:
df.head()

,Year,Title,IMDb,Viewership,Directors/Creators,Lead Actors,Release Date,Duration,Country,Genre
0,2016,Orange Is the New Black,8.0,23.0M Streams,Jenji Kohan,"Taylor Schilling, Uzo Aduba",17-Jun-16,S4; 730 min,USA,Comedy Drama
1,2016,Stranger Things,8.7,21.7M Streams,The Duffer Brothers,"Winona Ryder, Millie Bobby Brown",15-Jul-16,S1; 400 min,USA,Sci-Fi / Horror
2,2016,Fuller House,6.7,21.5M Streams,Jeff Franklin,"Candace Cameron Bure, Jodie Sweetin",25-Feb-16,S1; 390 min,USA,Sitcom / Comedy
3,2016,Marvel's Luke Cage,7.3,12.1M Streams,Cheo Hodari Coker,"Mike Colter, Alfre Woodard",30-Sep-16,S1; 715 min,USA,Action / Superhero
4,2016,Marvel's Daredevil,8.6,11.6M Streams,Drew Goddard,"Charlie Cox, Deborah Ann Woll",18-Mar-16,S2; 650 min,USA,Superhero


In [5]:
df.dtypes

Year                    int64
Title                  object
IMDb                  float64
Viewership             object
Directors/Creators     object
Lead Actors            object
Release Date           object
Duration               object
Country                object
Genre                  object
dtype: object

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 10 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Year                200 non-null    int64  
 1   Title               200 non-null    object 
 2   IMDb                200 non-null    float64
 3   Viewership          200 non-null    object 
 4   Directors/Creators  200 non-null    object 
 5   Lead Actors         200 non-null    object 
 6   Release Date        200 non-null    object 
 7   Duration            200 non-null    object 
 8   Country             200 non-null    object 
 9   Genre               200 non-null    object 
dtypes: float64(1), int64(1), object(8)
memory usage: 15.8+ KB


Tratamento e Limpeza (Transform) 

1.	Tratamento de Visualizações (Viewership):
    o	Remova o texto "M Streams" da coluna, deixando apenas os números.

    o	Converta a coluna para decimal (float).

    o	Renomeie a coluna para Viewership_Millions.

2.	Tratamento de Duração (Duration):
    o	O texto está no formato "S4; 730 min".

    o	Eu quero que você divida isso em duas colunas novas:
       
        	Uma coluna chamada Season contendo apenas a temporada (ex: "S4").
        
        	Uma coluna chamada Duration_Minutes contendo apenas os minutos em formato de número inteiro (ex: 730).


In [7]:
# removendo o texto e deixando apenas os números
df['Viewership'] = df['Viewership'].str.extract(r'([\d\.]+)', expand=False) 

In [8]:
df.head()

,Year,Title,IMDb,Viewership,Directors/Creators,Lead Actors,Release Date,Duration,Country,Genre
0,2016,Orange Is the New Black,8.0,23.0,Jenji Kohan,"Taylor Schilling, Uzo Aduba",17-Jun-16,S4; 730 min,USA,Comedy Drama
1,2016,Stranger Things,8.7,21.7,The Duffer Brothers,"Winona Ryder, Millie Bobby Brown",15-Jul-16,S1; 400 min,USA,Sci-Fi / Horror
2,2016,Fuller House,6.7,21.5,Jeff Franklin,"Candace Cameron Bure, Jodie Sweetin",25-Feb-16,S1; 390 min,USA,Sitcom / Comedy
3,2016,Marvel's Luke Cage,7.3,12.1,Cheo Hodari Coker,"Mike Colter, Alfre Woodard",30-Sep-16,S1; 715 min,USA,Action / Superhero
4,2016,Marvel's Daredevil,8.6,11.6,Drew Goddard,"Charlie Cox, Deborah Ann Woll",18-Mar-16,S2; 650 min,USA,Superhero


In [9]:
# convertendo a coluna para o tipo float
df['Viewership'] = df['Viewership'].astype(float)

In [11]:
# renomeando a coluna
df = df.rename(columns={'Viewership': 'Viewership_Millions'})

In [13]:
# revomendo as temporadas da coluna "Duration" e inserindo em uma nova
df ['Season'] = df ['Duration'].str.extract(r'([^\;]+)')

In [14]:
df.head()

,Year,Title,IMDb,Viewership_Millions,Directors/Creators,Lead Actors,Release Date,Duration,Country,Genre,Season
0,2016,Orange Is the New Black,8.0,23.0,Jenji Kohan,"Taylor Schilling, Uzo Aduba",17-Jun-16,S4; 730 min,USA,Comedy Drama,S4
1,2016,Stranger Things,8.7,21.7,The Duffer Brothers,"Winona Ryder, Millie Bobby Brown",15-Jul-16,S1; 400 min,USA,Sci-Fi / Horror,S1
2,2016,Fuller House,6.7,21.5,Jeff Franklin,"Candace Cameron Bure, Jodie Sweetin",25-Feb-16,S1; 390 min,USA,Sitcom / Comedy,S1
3,2016,Marvel's Luke Cage,7.3,12.1,Cheo Hodari Coker,"Mike Colter, Alfre Woodard",30-Sep-16,S1; 715 min,USA,Action / Superhero,S1
4,2016,Marvel's Daredevil,8.6,11.6,Drew Goddard,"Charlie Cox, Deborah Ann Woll",18-Mar-16,S2; 650 min,USA,Superhero,S2


In [15]:
df['Duration_Minutes'] = df['Duration'].str.extract(r';\s*(\d+)\s*min') # adicionando uma nova coluna com apenas os números

In [16]:
df['Duration_Minutes'] = df['Duration_Minutes'].astype(int) # convertendo eles para números inteiros

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Year                 200 non-null    int64  
 1   Title                200 non-null    object 
 2   IMDb                 200 non-null    float64
 3   Viewership_Millions  199 non-null    float64
 4   Directors/Creators   200 non-null    object 
 5   Lead Actors          200 non-null    object 
 6   Release Date         200 non-null    object 
 7   Duration             200 non-null    object 
 8   Country              200 non-null    object 
 9   Genre                200 non-null    object 
 10  Season               200 non-null    object 
 11  Duration_Minutes     200 non-null    int64  
dtypes: float64(2), int64(2), object(8)
memory usage: 18.9+ KB


3.	Tratamento de Datas:
    o	Converta a coluna de data (ex: "17-Jun-16") para o formato oficial de data (datetime) usando pd.to_datetime().


In [18]:
df ['Release Date'] = pd.to_datetime( df['Release Date'], format= '%d-%b-%y') # formatando as datas

4.	Padronização de Texto (Genre):
    o	Faça um .str.strip() na coluna Genre para remover os espaços em branco que vieram sobrando no final das palavras (como em "Comedy Drama ").


In [19]:
df ['Genre'] = df['Genre'].str.strip()

In [21]:
df = df.drop(columns=['Duration']) # removendo a coluna não formatada

In [22]:
df.head()

,Year,Title,IMDb,Viewership_Millions,Directors/Creators,Lead Actors,Release Date,Country,Genre,Season,Duration_Minutes
0,2016,Orange Is the New Black,8.0,23.0,Jenji Kohan,"Taylor Schilling, Uzo Aduba",2016-06-17,USA,Comedy Drama,S4,730
1,2016,Stranger Things,8.7,21.7,The Duffer Brothers,"Winona Ryder, Millie Bobby Brown",2016-07-15,USA,Sci-Fi / Horror,S1,400
2,2016,Fuller House,6.7,21.5,Jeff Franklin,"Candace Cameron Bure, Jodie Sweetin",2016-02-25,USA,Sitcom / Comedy,S1,390
3,2016,Marvel's Luke Cage,7.3,12.1,Cheo Hodari Coker,"Mike Colter, Alfre Woodard",2016-09-30,USA,Action / Superhero,S1,715
4,2016,Marvel's Daredevil,8.6,11.6,Drew Goddard,"Charlie Cox, Deborah Ann Woll",2016-03-18,USA,Superhero,S2,650


Fase 3: Regras de Negócio e Análise (Insights)

Com os dados perfeitamente limpos numéricos, extraia para o Diretor essas 3 informações (pode usar groupby, sort_values, etc):
   
    1.	O Maior Sucesso: Quais são os Top 5 títulos com o maior número de visualizações (Viewership_Millions)?
    
    2.	Qualidade Geográfica: Qual é a média da nota do IMDb agrupada por País (Country)? Quem produz as séries mais bem avaliadas?
   
    3.	A Maratona: Qual é o título (ou títulos) que tem a maior duração total em minutos (Duration_Minutes) da nossa base?


In [23]:
top_cinco_title = df.sort_values(by='Viewership_Millions', ascending=False) # ordenando os valores da coluna de forma decrescente

In [24]:
df = top_cinco_title

In [25]:
df[['Title', 'Viewership_Millions']].head(5) # exibindo as cinco primeiras linhas

,Title,Viewership_Millions
104,Money Heist,792.0
123,All of Us Are Dead,679.0
126,Bridgerton 2,656.0
83,Money Heist,619.0
61,Stranger Things 3,582.0


In [26]:
media_por_pais = df.groupby('Country')['IMDb'].mean() # usando o groupby para agrupar os valores de cada país para depois tirar a média com o mean()

In [27]:
print(media_por_pais)

Country
Australia/USA    7.200000
Brazil           7.300000
Canada           8.633333
Canada/USA       8.000000
Colombia         5.600000
Denmark          7.500000
France           6.333333
Germany          7.600000
Japan            7.700000
Mexico           6.900000
Norway           6.500000
South Korea      7.895000
Spain            7.455556
Sweden           7.300000
UK               8.139130
UK/Canada        8.600000
UK/Italy         6.800000
UK/USA           8.350000
USA              7.635652
USA/Colombia     8.800000
USA/France       7.950000
USA/Japan        7.050000
USA/Mexico       8.400000
USA/Poland       8.000000
USA/UK           7.600000
Name: IMDb, dtype: float64


In [28]:
pais_campeao = media_por_pais.idxmax() # retorna o INDICIE do valor mais alto

In [29]:
mais_avaliado = media_por_pais.max() # retorna o VALOR mais alto

In [30]:
print(f'O país mais bem avaliado é: {pais_campeao} com {mais_avaliado} de IMDb')

O país mais bem avaliado é: USA/Colombia com 8.8 de IMDb


In [31]:
top_title_duration = df.sort_values(by='Duration_Minutes', ascending=False)[['Title', 'Duration_Minutes']].head(1) # agrupar os valores e deixando de forma decrescente e depois exibindo a primeira linha com o head(1)

top_title_duration

,Title,Duration_Minutes
137,Alchemy of Souls,1800


In [32]:
df = df.sort_values(by='Year', ascending=False) # ordenar o arquivo pelo ano

In [33]:
df.head()

,Year,Title,IMDb,Viewership_Millions,Directors/Creators,Lead Actors,Release Date,Country,Genre,Season,Duration_Minutes
183,2025,Wednesday 2,8.1,124.0,Tim Burton,Jenna Ortega,2025-08-06,USA,Supernatural Mystery,S2,467
191,2025,Gabby's Dollhouse,7.1,108.0,Traci Paige Johnson,Laila Lockhart,2025-07-01,USA,Kids / Family,All Seasons,1000
188,2025,A House of Dynamite,7.2,76.0,Kathryn Bigelow,Idris Elba,2025-08-20,USA,Drama,S1,400
187,2025,UNTAMED,7.8,93.0,Mark L. Smith,Eric Bana,2025-07-17,USA,Crime Thriller,S1,300
181,2025,Squid Game 3,8.0,145.8,Hwang Dong-hyuk,Lee Jung-jae,2025-06-27,South Korea,Survival Drama,S3,368


In [35]:
df.to_csv('netflix_limpa_final.csv', index=False, sep=';', encoding='utf-8-sig')